enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,701 @@
+* Increases support complexity due to integration with the a user's provider network.
+* Increases network control plane load on nodes, as they have multiple peer connections and are having to maintain
+a dynamic routing protocol.
+
+### Implementation Details/Notes/Constraints
+
+#### BGP Peering
+
+FRR-K8S shall support both internal BGP (iBGP) and external BGP (eBGP). With iBGP, a full mesh topology is required. This
+can be cumbersome and have issues with scaling in large clusters. In the future, FRR-K8S shall also support designating one
+or more nodes as route reflectors (https://issues.redhat.com/browse/CNF-10719). With route reflectors, nodes only have
+to connect to the route reflectors and do not have to maintain a full mesh with all other nodes in the cluster. Therefore,
+support for iBGP in this enhancement will be restricted to full mesh only.
+
+#### FRR-K8S Integration
+
+As previously mentioned frr-k8s will be used in order to deploy and manage FRR configuration. Support will be added
+where necessary for gaps in frr-k8s API, but in the meantime frr-k8s supports also issuing raw FRR configuration. Every
+attempt should be made to leverage the FRR-K8S API, and raw configuration should only be used as a last resort. A user
+will need the ability to:
+
+* Define BGP peers, ASN, authentication, along with other common BGP configurations like communities, route filtering, etc.
+* Configure BFD
+
+FRR-K8S will serve as the main API for user configuration of BGP via FRR-K8S CRs. OVN-Kubernetes will watch for these
+CRs, and once created, OVN-Kubernetes will start monitoring the kernel routing table via netlink for routes installed
+by FRR. When FRR is using Zebra, the routes can be identified via a proto "bgp" label:
+
+```
+123.123.123.0/24 nhid 47 via 172.18.0.5 dev breth0 proto bgp metric 20
+```
+
+When these routes are seen on the node, ovnkube-controller will update the routes in NBDB for any GW routers on the
+node, filtering the route for the GW router based on if the interface on the route matches the interface in OVS. This
+will happen irrespective of gateway mode, as some features in local gateway mode rely on the GRs for sending traffic
+(Egress IP, MEG). Note, this configuration will only happen on nodes selected by the FRR-K8S CRs.
+
+#### API Changes
+
+##### OpenShift API
+
+The OpenShift API will be modified in order to allow CNO to deploy FRR and FRR-K8S CRDs:
+
+```golang
+// NetworkSpec is the top-level network configuration object.
+type NetworkSpec struct {
+    // ...
+    // deployFRR specifies whether or not the Free Range Routing (FRR) stack
+    // along with FRR-K8S API should be deployed by the operator.
+    // FRR is required for enabling for using any OpenShift features that require dynamic
+    // routing. This includes BGP support for OVN-Kubernetes and MetalLB.
+    // +optional
+    DeployFRR *bool `json:"deployFRR,omitempty"`
```

Comment:
Seems like this should have a more generic name referring to "BGP" rather than referring to the specific implementation of it that we happen to currently be using.

Reply:
Yeah Jaime had the same feedback. I didn't want to list is as "BGP" because FRR is an entire routing stack itself, so it would also enable OSPF, etc. I think @jcaamano came up with "AdvancedRouting" so I'll go with that for now.